lib/orderbook/OrderBook.ts

```diff
@@ -80,7 +80,18 @@ class OrderBook {
     const { matches, remainingOrder } = matchingResult;
 
     if (matches.length > 0) {
-      matches.forEach(this.handleMatch);
+      for (const [index, match] of matches.entries()) {
+        const { maker, taker } = match;
+        this.handleMatch({ maker, taker });
+
+        // Maker is fully matched if it is not the last
+        if ((matches.length - index) > 1) {
```

Comment:
although that's correct, it's better to not make assumptions about `MatchingEngine` here, and to treat all matches in the same way.